src/Tutorial_Ltac2_backtracking.v

```
@@ -0,0 +1,576 @@
+    We can check it works using [constructor].
+*)
+
+Goal nat -> False -> True \/ False.
+  intros. constructor; print_goals; my_eassumption.
+Qed.
+
+(** You may be accustomed to use [fail], for example, if you are a Ltac1 user.
+    In Ltac2, [fail] is easy to understand as it is literaly defined as
+    [Control.zero (Tactic_failure None)].
+    Therefore, [fail] just is just failure raising the error [Tactic Failure]
+    without any further error message.
+    It is recommended to use [Control.zero] rather than [fail] to give nice
+    error messages in case of failure.
+*)
+
+Goal nat.
+  Fail fail.
+Abort.
+
+(** There is another primitive to raise exceptions [Control.throw : exn -> unit].
+    The key difference with [zero] is that [throw] raises a non-catchable exception.
+    It means that [throw] will not trigger backtracking.
+    It will stop the computation all together.
+*)
+
+Goal nat -> False -> True \/ False.
+  intros.
+  (* This will not backtrack and try to prove [False] *)
+  Fail constructor; print_goals; Control.throw (Tactic_failure None).
+Abort.
+
+(** Consequently, [throw] should only be used if one **really* wants to prevent backtracking. *)
+
+
+
+(** ** 3. [Control.plus] to Catch Exceptions
+
+    Since [Ltac2] has proper exceptions and backtracking is primitive, it is not
+    only possible to catch exceptions, but also to backtrack upon them.
+
+    The catching primitive is [Control.plus] of type [(unit -> 'a) -> (exn -> 'a) -> 'a].
+    [Control.plus f k] installs a backtracking point around [f]:
+    1. it tries all the success of [f], and if there are none, it tries [k]
+    2. in case of subsequent failure, it backtracks to [f], and if all its
+       successes have been exhausted, it tries the continuation [k] for next success.
+
+    To understand [Control.plus], it can be useful to see it from the perspective of the stream model.
+    Viewing [f] and [k] as stream of success -- ignoring exceptions for simplicity --
```

I don't think you can say that you can ignore exceptions for simplicity, they are the basic block of the backtracking model.